### ISSUE TITLE
switch_case_alignment doesn't work with switch expressions

### ISSUE THREAD

**Issue by futuretap** [realm/SwiftLint#5080]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Describe the bug

A switch expression that is assigned to a variable ([SE0380](https://github.com/apple/swift-evolution/blob/main/proposals/0380-if-switch-expressions.md)) throws the error 

`warning:  Switch and Case Statement Alignment Violation: Case statements should vertically align with their enclosing switch statement (switch_case_alignment)`

##### Complete output when running SwiftLint, including the stack trace and command used

see above

### Environment

* SwiftLint version 0.52.3
* CocoaPods
* No configuration file used (default CocoaPods installation)
* Xcode version 15.0 beta 2 (15A5161b)
* Do you have a sample that shows the issue?

```swift
// This triggers a violation:
let color: UIColor = switch state {
case [.highlighted]:
	.gray
case .selected, [.selected, .highlighted]:
	.white
case .disabled:
	.lightGray
default:
	.clear
}
```

**Comment by SimplyDanny** [comment]
I've "fixed" it by disabling this rule for such cases for the time being. Would you let me know how you see this issue to be fixed properly in #8086, @futuretap?

**Comment by futuretap** [comment]
That's what I did too. I'm confused about your question though. A proper fix would be that swiftlint accepts the mentioned code snippet. The vertical alignment of `case` and `default` should refer to the closing curly brace.

**Comment by SimplyDanny** [comment]
> That's what I did too. I'm confused about your question though. A proper fix would be that swiftlint accepts the mentioned code snippet. The vertical alignment of `case` and `default` should refer to the closing curly brace.

Ah sorry ... my comment was very confusing. I mean that I prepared a [PR](https://github.com/realm/SwiftLint/pull/5086) ignoring `switch` expressions assigned to variables in the rule. I wonder if this is what you'd expect. The rule has a configuration which could as well be used to configure it's behavior for `switch` expression really used as such.

**Comment by futuretap** [comment]
Thanks, this works well for me.
